# <span style="font-size: 20pt; font-family: Montserrat, sans-serif; color: rgb(92, 148, 206); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Graded Assignment 3: 9 to 5</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to show off your SQL skills! For each question, copy the SQL query you used and make note of the answer.</span>

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The Dataset</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">For this assignment, you will be using the Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results which can be found on</span> [<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(17, 85, 204); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; text-decoration-line: underline; vertical-align: baseline; white-space: pre-wrap;">Kaggle</span>](https:\www.kaggle.com\bls\employment)<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">.</span>

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Business Issue</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">You are working for the Bureau of Labor Statistics with the United States government and have been approached by your boss with an important meeting request. You have been asked by your supervisor to meet with Dolly Parton whose nonprofit is looking to shed light on the state of employment in the United States. As part of the 9 to 5 project, their research is focused on production and nonsupervisory employees and how those employees fare compared to all employees in the United States. While the data the BLS collects from the CES is publicly available, Dolly Parton and her colleagues need your assistance navigating the thousands of rows in each table in LaborStatisticsDB.</span>

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">About the Dataset</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">This dataset comes directly from the Bureau of Labor Statistics’ Current Employment Survey (CES). Here are some things you need to know:</span>

1. <span style="font-size: 10pt; background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The industry table contains an NAICS code. This is different from the industry code. NAICS stands for North American Industry Classification System.</span>
    
2. <span style="font-size: 10pt; background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Series ID is composed of multiple different codes. CES stands for Current Employment Survey, the name of the survey which collected the data. The industry code as specified by the BLS and the data type code as specified in the datatype table.</span>
    

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Set Up</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">To connect to the database, use the same connection info used during the SQL lessons.&nbsp;</span> 

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">For the assignment, we will be using the LaborStatisticsDB database.</span>

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Database Exploration</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">To start with, let’s get to know the database further.</span>

1. <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: &quot;Times New Roman&quot;; font-size: 10pt; white-space: pre-wrap;">Use this space to make note of each table in the database, the columns within each table, each column’s data type, and how the tables are connected. You can write this down or draw a diagram. Whatever method helps you get an understanding of what is going on with LaborStatisticsDB.</span>

```
flow.png

```

In [165]:
/* SCRATCH PAD */

/*USE LaborStatisticsDB
SELECT a.id, s.series_id, s.series_title, a.year, a.period, a.value, i.industry_name, ss.supersector_name
FROM dbo.annual_2016 a

LEFT JOIN dbo.series s ON s.series_id = a.series_id
LEFT JOIN dbo.supersector ss ON ss.supersector_code = s.supersector_code
LEFT JOIN dbo.industry i ON i.industry_code = s.industry_code
LEFT JOIN dbo.datatype d ON d.data_type_code = s.data_type_code
*/

/*USE LaborStatisticsDB
SELECT * 
FROM dbo.datatype

ORDER BY data_type_code */

USE LaborStatisticsDB
SELECT DISTINCT series_id, series_title
from dbo.series

ORDER BY series_title

(26709 rows affected)

Displaying Top 5000 rows.

Total execution time: 00:00:00.670

series_id series_title CES0500000056 Aggregate weekly hours of all employees CES0600000056 Aggregate weekly hours of all employees CES0800000056 Aggregate weekly hours of all employees CES1000000056 Aggregate weekly hours of all employees CES1011330056 Aggregate weekly hours of all employees CES1021000056 Aggregate weekly hours of all employees CES1021100056 Aggregate weekly hours of all employees CES1021200056 Aggregate weekly hours of all employees CES1021210056 Aggregate weekly hours of all employees CES1021220056 Aggregate weekly hours of all employees CES1021230056 Aggregate weekly hours of all employees CES1021231056 Aggregate weekly hours of all employees CES1021231256 Aggregate weekly hours of all employees CES1021231956 Aggregate weekly hours of all employees CES1021300056 Aggregate weekly hours of all employees CES1021311256 Aggregate weekly hours of all employees CES2000000056 Aggregate weekly hours of all employees CES2023600056 Aggregate weekly hours of all employees CES2023610056 Aggregate weekly hours of all employees CES2023611556 Aggregate weekly hours of all employees CES2023611856 Aggregate weekly hours of all employees CES2023620056 Aggregate weekly hours of all employees CES2023621056 Aggregate weekly hours of all employees CES2023622056 Aggregate weekly hours of all employees CES2023700056 Aggregate weekly hours of all employees CES2023710056 Aggregate weekly hours of all employees CES2023711056 Aggregate weekly hours of all employees CES2023712056 Aggregate weekly hours of all employees CES2023713056 Aggregate weekly hours of all employees CES2023720056 Aggregate weekly hours of all employees CES2023730056 Aggregate weekly hours of all employees CES2023790056 Aggregate weekly hours of all employees CES2023800056 Aggregate weekly hours of all employees CES2023810056 Aggregate weekly hours of all employees CES2023811056 Aggregate weekly hours of all employees CES2023812056 Aggregate weekly hours of all employees CES2023813056 Aggregate weekly hours of all employees CES2023814056 Aggregate weekly hours of all employees CES2023815056 Aggregate weekly hours of all employees CES2023816056 Aggregate weekly hours of all employees CES2023817056 Aggregate weekly hours of all employees CES2023819056 Aggregate weekly hours of all employees CES2023820056 Aggregate weekly hours of all employees CES2023821056 Aggregate weekly hours of all employees CES2023822056 Aggregate weekly hours of all employees CES2023829056 Aggregate weekly hours of all employees CES2023830056 Aggregate weekly hours of all employees CES2023831056 Aggregate weekly hours of all employees CES2023832056 Aggregate weekly hours of all employees CES2023833056 Aggregate weekly hours of all employees CES2023834056 Aggregate weekly hours of all employees CES2023835056 Aggregate weekly hours of all employees CES2023839056 Aggregate weekly hours of all employees CES2023890056 Aggregate weekly hours of all employees CES2023891056 Aggregate weekly hours of all employees CES2023899056 Aggregate weekly hours of all employees CES3000000056 Aggregate weekly hours of all employees CES3100000056 Aggregate weekly hours of all employees CES3132100056 Aggregate weekly hours of all employees CES3132110056 Aggregate weekly hours of all employees CES3132120056 Aggregate weekly hours of all employees CES3132190056 Aggregate weekly hours of all employees CES3132191056 Aggregate weekly hours of all employees CES3132191156 Aggregate weekly hours of all employees CES3132191856 Aggregate weekly hours of all employees CES3132192056 Aggregate weekly hours of all employees CES3132199056 Aggregate weekly hours of all employees CES3132700056 Aggregate weekly hours of all employees CES3132730056 Aggregate weekly hours of all employees CES3132732056 Aggregate weekly hours of all employees CES3132739056 Aggregate weekly hours of all employees CES3132790056 Aggregate weekly hours of all employees CES3133100056 Aggregate weekly hours of all employees CES3133120056 Aggregate weekly hour

2. <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: Arvo, serif; font-size: 10pt; white-space: pre-wrap;">What is the datatype for women employees?</span>

In [149]:
USE LaborStatisticsDB
SELECT * 
FROM dbo.datatype

/* WHERE data_type_text LIKE '%women employees%' */
WHERE data_type_text = 'women employees'

(1 row affected)

Total execution time: 00:00:00.058

data_type_code data_type_text 10 WOMEN EMPLOYEES

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the series id for&nbsp; women employees in the commercial banking industry in the financial activities supersector?</span>

In [150]:
USE LaborStatisticsDB
SELECT s.series_id, s.series_title, i.industry_code, i.industry_name, ss.supersector_code, ss.supersector_name
FROM dbo.series s

JOIN dbo.industry i ON i.industry_code = s.industry_code
JOIN dbo.supersector ss ON ss.supersector_code = s.supersector_code

WHERE s.series_title = 'women employees' AND i.industry_name LIKE '%commercial banking%' AND ss.supersector_name LIKE '%financial activities%'

(2 rows affected)

Total execution time: 00:00:00.082

series_id series_title industry_code industry_name supersector_code supersector_name CES5552211010 Women employees 55522110 Commercial banking 55 Financial activities CEU5552211010 Women employees 55522110 Commercial banking 55 Financial activities

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Join in on the Fun</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to start joining! You can choose the type of join you use, just make sure to make a&nbsp; note!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join&nbsp; annual_2016 with series on series_id. We only want the data in the annual_2016 table to be included in the result.</span>

In [151]:
USE LaborStatisticsDB
SELECT TOP 15 a.*
FROM dbo.annual_2016 a

LEFT JOIN dbo.series s ON a.series_id = s.series_id

(15 rows affected)

Total execution time: 00:00:00.114

id series_id year period value footnote_codes original_file 0 CEU5500000007 2016 M13 36.9 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 1 CEU5500000008 2016 M13 26.11 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 2 CEU5500000030 2016 M13 962.73 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 3 CEU5500000031 2016 M13 411.29 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 4 CEU5500000032 2016 M13 11.15 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 5 CEU5500000034 2016 M13 111.6 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 6 CEU5500000035 2016 M13 179.2 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 7 CEU5500000081 2016 M13 236997 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 8 CEU5500000082 2016 M13 6189003 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 9 CEU5552200007 2016 M13 37.3 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 10 CEU5552200008 2016 M13 22.37 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 11 CEU5552200030 2016 M13 833.43 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 12 CEU5552200031 2016 M13 356.05 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 13 CEU5552200032 2016 M13 9.56 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 14 CEU5552200034 2016 M13 104.2 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and datatype on data_type_code</span>

In [152]:
USE LaborStatisticsDB
SELECT TOP 15 *
FROM dbo.series s

JOIN dbo.datatype d ON d.data_type_code = s.data_type_code

(15 rows affected)

Total execution time: 00:00:00.072

series_id supersector_code industry_code data_type_code seasonal series_title data_type_code data_type_text CES0000000001 0 00000000 1 S All employees 1 ALL EMPLOYEES CES0000000010 0 00000000 10 S Women employees 10 WOMEN EMPLOYEES CES0000000025 0 00000000 25 S All employees 25 ALL EMPLOYEES CES0000000026 0 00000000 26 S All employees 26 ALL EMPLOYEES CES0500000001 5 05000000 1 S All employees 1 ALL EMPLOYEES CES0500000002 5 05000000 2 S Average weekly hours of all employees 2 AVERAGE WEEKLY HOURS OF ALL EMPLOYEES CES0500000003 5 05000000 3 S Average hourly earnings of all employees 3 AVERAGE HOURLY EARNINGS OF ALL EMPLOYEES CES0500000006 5 05000000 6 S Production and nonsupervisory employees 6 PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000007 5 05000000 7 S Average weekly hours of production and nonsupervisory employees 7 AVERAGE WEEKLY HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000008 5 05000000 8 S Average hourly earnings of production and nonsupervisory employees 8 AVERAGE HOURLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000010 5 05000000 10 S Women employees 10 WOMEN EMPLOYEES CES0500000011 5 05000000 11 S Average weekly earnings of all employees 11 AVERAGE WEEKLY EARNINGS OF ALL EMPLOYEES CES0500000012 5 05000000 12 S Average weekly earnings of all employees 12 AVERAGE WEEKLY EARNINGS OF ALL EMPLOYEES CES0500000013 5 05000000 13 S Average hourly earnings of all employees 13 AVERAGE HOURLY EARNINGS OF ALL EMPLOYEES CES0500000016 5 05000000 16 S Indexes of aggregate weekly hours of all employees 16 INDEXES OF AGGREGATE WEEKLY HOURS OF ALL EMPLOYEES

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and industry on industry_code</span>

In [153]:
USE LaborStatisticsDB
SELECT TOP 15 *
FROM dbo.series s

JOIN dbo.industry i ON i.industry_code = s.industry_code

(15 rows affected)

Total execution time: 00:00:00.081

series_id supersector_code industry_code data_type_code seasonal series_title id industry_code naics_code publishing_status industry_name display_level selectable sort_sequence CES0000000001 0 00000000 1 S All employees 0 0 - B Total nonfarm 0 T 1 CES0000000010 0 00000000 10 S Women employees 0 0 - B Total nonfarm 0 T 1 CES0000000025 0 00000000 25 S All employees 0 0 - B Total nonfarm 0 T 1 CES0000000026 0 00000000 26 S All employees 0 0 - B Total nonfarm 0 T 1 CES0500000001 5 05000000 1 S All employees 1 5000000 - A Total private 1 T 2 CES0500000002 5 05000000 2 S Average weekly hours of all employees 1 5000000 - A Total private 1 T 2 CES0500000003 5 05000000 3 S Average hourly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000006 5 05000000 6 S Production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000007 5 05000000 7 S Average weekly hours of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000008 5 05000000 8 S Average hourly earnings of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000010 5 05000000 10 S Women employees 1 5000000 - A Total private 1 T 2 CES0500000011 5 05000000 11 S Average weekly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000012 5 05000000 12 S Average weekly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000013 5 05000000 13 S Average hourly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000016 5 05000000 16 S Indexes of aggregate weekly hours of all employees 1 5000000 - A Total private 1 T 2

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Aggregate Your Friends and Code some SQL</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Put together the following:</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp; No chopping people into little bits, please.</span>

In [154]:
USE LaborStatisticsDB
SELECT ROUND(SUM(a.value),0)
FROM dbo.annual_2016 a

LEFT JOIN dbo.series s ON a.series_id = s.series_id
INNER JOIN dbo.industry i ON i.industry_code = s.industry_code

WHERE s.series_title = 'All Employees'



(1 row affected)

Total execution time: 00:00:00.096

(No column name) 2340612

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many women employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp;</span>

In [155]:
USE LaborStatisticsDB
SELECT ROUND(SUM(a.value),0)
FROM dbo.annual_2016 a

LEFT JOIN dbo.series s ON a.series_id = s.series_id
INNER JOIN dbo.industry i ON i.industry_code = s.industry_code

WHERE s.series_title = 'Women Employees'

(1 row affected)

Total execution time: 00:00:00.085

(No column name) 1125490

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many production/nonsupervisory employees were reported in 2016? Round to the nearest whole number.&nbsp;</span>

In [156]:
USE LaborStatisticsDB
SELECT ROUND(SUM(a.value),0)
FROM dbo.annual_2016 a

LEFT JOIN dbo.series s ON a.series_id = s.series_id
INNER JOIN dbo.industry i ON i.industry_code = s.industry_code

WHERE s.series_title = 'Production and nonsupervisory employees'


(1 row affected)

Total execution time: 00:00:00.085

(No column name) 1263650

4. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2017, what is the average weekly hours worked by production and nonsupervisory employees across all industries?</span>

In [157]:
USE LaborStatisticsDB
SELECT ROUND(AVG(j.value),2)
FROM dbo.january_2017 j

LEFT JOIN dbo.series s ON j.series_id = s.series_id
INNER JOIN dbo.industry i ON i.industry_code = s.industry_code

WHERE s.series_title = 'Average weekly hours of production and nonsupervisory employees'

(1 row affected)

Total execution time: 00:00:00.100

(No column name) 36.06

5. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the total weekly payroll for production and nonsupervisory employees across all industries in January 2017? Round to the nearest penny, please.</span>

In [158]:
USE LaborStatisticsDB
SELECT ROUND(SUM(j.value),2)
FROM dbo.january_2017 j

LEFT JOIN dbo.series s ON j.series_id = s.series_id
INNER JOIN dbo.industry i ON i.industry_code = s.industry_code

WHERE s.series_title = 'AGGREGATE WEEKLY PAYROLLS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES'



(1 row affected)

Total execution time: 00:00:00.089

(No column name) 1838753220

6. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2017, which industry was the average weekly hours worked by production and nonsupervisory employees the highest? Which industry was the lowest?</span>

In [159]:
USE LaborStatisticsDB
SELECT i.industry_name,
 ROUND(AVG([value]),1) as actual_avg_hours
FROM dbo.january_2017 j

LEFT JOIN dbo.series s ON j.series_id = s.series_id
INNER JOIN dbo.industry i ON i.industry_code = s.industry_code

WHERE s.series_title = 'AVERAGE WEEKLY HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES'

group by i.industry_name 

/* HIGHEST */ ORDER BY actual_avg_hours desc
/* LOWEST *//*ORDER BY actual_avg_hours asc*/

(501 rows affected)

Total execution time: 00:00:00.148

industry_name actual_avg_hours Motor vehicle power train components 49.6 Support activities for mining 47.2 Support activities for oil and gas operations 46.7 Motor vehicle metal stamping 46.7 Coal mining 46.6 Oil and gas pipeline construction 46.1 Mining 46 Boilers 45.7 Crushed and broken limestone mining 45.7 Mining and logging 45.6 Stone mining and quarrying 45.5 Petroleum and coal products 45.4 Waste treatment and disposal 45.1 Basic chemicals 44.9 Motor vehicle parts 44.6 Rubber products 44.4 Pipeline transportation 44.3 Surgical appliances and supplies 44.1 Metal valves 44.1 All other motor vehicle parts 44 Turbine and power transmission equipment 44 Transportation equipment 43.8 Pulp 43.8 Frozen food 43.8 Motor vehicle gasoline engine and parts 43.7 Electric power transmission and distribution 43.6 Power and communication system construction 43.6 Power generation and supply 43.6 Other heavy construction 43.5 Special tools 43.5 Corrugated and solid fiber boxes 43.5 Aerospace products and parts 43.5 Animal slaughtering and processing 43.5 Fossil fuel electric power generation 43.5 Other specialized trucking 43.4 Other electrical equipment and components 43.3 Utility system construction 43.3 HVAC and commercial refrigeration equipment 43.2 Electric power generation 43.1 Oil and gas extraction 43.1 Paperboard containers 43.1 Utilities 43 Forging and stamping 43 Heavy and civil engineering construction 42.9 Metalworking machinery 42.8 Machinery 42.8 Electrical equipment and appliances 42.8 Turned products and screws 42.8 Petroleum 42.8 Sawmills and wood preservation 42.8 Other general purpose machinery 42.6 Fruit and vegetable preserving and specialty 42.6 Electrical equipment 42.5 Aircraft 42.5 Other fabricated metal products 42.5 Primary metals 42.5 Plastics and rubber products 42.5 Agricultural 42.4 Construction equipment 42.4 Highway 42.4 Cement and concrete products 42.3 Poultry processing 42.2 Plastics packaging materials 42.1 Converted paper products 42.1 Machine shops and threaded products 42.1 Medical equipment and supplies 42.1 Fruits and vegetables 42 Fabricated metal products 42 Plastics products 42 Resin 42 Packaging and labeling services 42 Electronic instruments 41.9 Industrial building 41.9 Industrial machinery 41.9 Machine shops 41.9 Manufacturing 41.8 Millwork 41.8 Other building equipment contractors 41.8 Wood products 41.8 Ornamental and architectural metal products 41.7 All other fabricated metal products 41.7 General freight trucking 41.6 Other plastics products 41.6 Plywood and engineered wood products 41.6 Semiconductors and related devices 41.5 Other wood products 41.5 Wood containers and pallets 41.5 General warehousing and storage 41.5 Fruit and vegetable canning and drying 41.5 Heavy machinery rental and leasing 41.5 Natural gas distribution 41.5 Lumber and wood 41.5 Architectural and structural metals 41.5 Logging 41.4 Nonmetallic mineral products 41.4 Food manufacturing 41.4 Foundries 41.4 Wood kitchen cabinets and countertops 41.4 Warehousing and storage 41.4 Sheet metal work 41.4 Plate work and fabricated structural products 41.4 Truck transportation 41.3 Commercial and service industry machinery 41.3 Fabricated structural metal products 41.2 Dairy products 41.2 Water and sewer system construction 41.2 Water 41.1 Waste management and remediation services 41.1 Wired telecommunications carriers 41 Computer and electronic products 41 Lumber and construction supplies 41 Motor vehicles 40.9 Durable goods 40.9 Coating 40.9 Chemicals 40.9 All other wood products 40.9 Recyclable materials 40.9 Paper and paper products 40.8 Direct insurers 40.8 Goods-producing 40.8 Motor vehicle bodies and trailers 40.7 Electrical equipment and wiring 40.7 Specialized freight trucking 40.6 Textile mills 40.6 Paper bags and coated and treated paper 40.5 Semiconductors and electronic components 40.5 Tile and terrazzo contractors 40.4 Telecommunications 40.3 Soaps 40.3 Miscellaneous durable goods manufacturing 40.3 Furnit

7. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2021, which industry was the total weekly payroll for production and nonsupervisory employees the highest? Which industry was the lowest?</span>

In [160]:
USE LaborStatisticsDB
SELECT i.industry_name, ROUND(SUM([value]),1) as total_payroll
FROM dbo.january_2017 j

LEFT JOIN dbo.series s ON j.series_id = s.series_id
INNER JOIN dbo.industry i ON i.industry_code = s.industry_code

WHERE s.series_title = 'AGGREGATE WEEKLY PAYROLLS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES'

group by i.industry_name 

/* HIGHEST */ ORDER BY total_payroll desc
/* LOWEST *//*ORDER BY total_payroll asc*/

(501 rows affected)

Total execution time: 00:00:00.163

industry_name total_payroll Total private 295944946 Private service-providing 243444986 NULL 69500056 Professional and business services 60942150 Trade 59165538 Education and health services 59162460 Goods-producing 52277530 Health care and social assistance 51281976 Professional and technical services 35595426 Manufacturing 29807366 Durable goods 28424368 Financial activities 25592788 Retail trade 24367308 Hospitals 20307536 Construction 20118860 Ambulatory health care services 19566350 Administrative and waste services 19091232 General medical and surgical hospitals 18980922 Wholesale trade 17972488 Administrative and support services 17846870 Leisure and hospitality 17522542 Offices of physicians 17050766 Nondurable goods 15908492 Accommodation and food services 14750008 Transportation and warehousing 14068472 Specialty trade contractors 12968226 Food services and drinking places 11997448 Other services 11424024 Computer systems design and related services 10783198 Restaurants and other eating places 10772160 Information 9702870 Insurance carriers and related activities 9511636 General freight trucking 8744250 Employment services 7903810 Credit intermediation and related activities 6970098 Building equipment contractors 6634692 Membership associations and organizations 6389542 Full-service restaurants 6311336 Management of companies and enterprises 6307426 Temporary help services 6259442 Nursing and residential care facilities 6030332 Managing offices 6027242 Insurance carriers 5975556 Architectural and engineering services 5724824 Social assistance 5389302 Management and technical consulting services 5334362 Computer systems design services 5167844 Transportation equipment 5000898 Custom computer programming services 4743554 Motor vehicle and parts dealers 4687434 Truck transportation 4448496 Legal services 4305520 Management consulting services 4240586 General merchandise stores 4231560 Offices of lawyers 4084946 Depository credit intermediation 3951730 Construction of buildings 3897670 Food and beverage stores 3876050 Limited-service restaurants 3783578 Insurance agencies 3534552 Fabricated metal products 3477080 Grocery stores 3459308 Electronic markets and agents and brokers 3457738 Real estate 3457214 Food manufacturing 3387214 Wholesale trade agents and brokers 3365740 Automobile dealers 3341452 Individual and family services 3335314 Supermarkets and other grocery stores 3277380 Heavy and civil engineering construction 3258356 Nursing care facilities 3206008 Services to buildings and dwellings 3124638 Plumbing and HVAC contractors 3093382 Direct life and health insurance carriers 3086720 Scientific research and development services 3082548 Electrical contractors 3044426 Commercial banking 3023714 New car dealers 2999190 Telecommunications 2989914 Research and development in the physical 2930342 Repair and maintenance 2921080 Publishing industries 2901224 Mining 2899614 Arts 2816804 Utilities 2760324 Direct insurers 2752454 Accommodation 2733694 Traveler accommodation 2676012 Home health care services 2670246 Accounting and bookkeeping services 2582570 Outpatient care centers 2579572 Machinery 2570424 Computer and electronic products 2407066 Chemicals 2397810 Commercial equipment 2377234 Building foundation and exterior contractors 2371776 Administrative management consulting services 2365312 Insurance agencies and brokerages 2364092 Mining and logging 2362250 Nonresidential building 2254214 Offices of dentists 2242014 Warehousing and storage 2238694 Physical 2236444 Services for the elderly and persons with disabilities 2227556 Wired telecommunications carriers 2222846 Hotels and motels 2207690 Building finishing contractors 2170306 Personal and laundry services 2104422 Power generation and supply 2094130 Machinery and supplies 2079612 Other financial investment activities 2071800 Building material and garden supply stores 2047090 Nondepository credit intermediation 2030996 Grocery and related products 2019294 Genera

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Subqueries, Unions, Derived Tables, Oh My!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Write a query that returns the series_id, industry_code, industry_name, and value from the january_2017 table but only if that value is greater than the average value for annual_2016 of data_type_code 82.</span>

In [161]:
USE LaborStatisticsDB
SELECT s.series_id, i.industry_code, i.industry_name, j.value
FROM dbo.january_2017 j

LEFT JOIN dbo.series s ON s.series_id = j.series_id
LEFT JOIN dbo.industry i ON i.industry_code = s.industry_code

WHERE j.value >
    (SELECT round(avg(a.value),2)
    FROM dbo.annual_2016 a

    LEFT JOIN dbo.series s ON a.series_id = s.series_id
    LEFT JOIN dbo.datatype d ON s.data_type_code = d.data_type_code
    WHERE d.data_type_code = 82)

ORDER BY j.value

/* Checking - 819960.34 */
/*SELECT round(avg(a.value),2)
    FROM dbo.annual_2016 a

    LEFT JOIN dbo.series s ON a.series_id = s.series_id
    LEFT JOIN dbo.datatype d ON s.data_type_code = d.data_type_code
    WHERE d.data_type_code = 82 */

(754 rows affected)

Total execution time: 00:00:00.249

series_id industry_code industry_name value CES4244511082 42445110 Supermarkets and other grocery stores 821282 CES4244511082 42445110 Supermarkets and other grocery stores 821282 CES6562410082 65624100 Individual and family services 828771 CES6562410082 65624100 Individual and family services 828771 CEU4244110082 42441100 Automobile dealers 831131 CEU4244110082 42441100 Automobile dealers 831131 CES4142512082 41425120 Wholesale trade agents and brokers 831319 CES4142512082 41425120 Wholesale trade agents and brokers 831319 CES2023810057 20238100 Building foundation and exterior contractors 833625 CES2023810057 20238100 Building foundation and exterior contractors 833625 CEU6562410082 65624100 Individual and family services 838886 CEU6562410082 65624100 Individual and family services 838886 CES6562140057 65621400 Outpatient care centers 839455 CES6562140057 65621400 Outpatient care centers 839455 CES4244110082 42441100 Automobile dealers 839595 CES4244110082 42441100 Automobile dealers 839595 CEU3231100082 32311000 Food manufacturing 842201 CEU3231100082 32311000 Food manufacturing 842201 CES6562120057 65621200 Offices of dentists 842558 CES6562120057 65621200 Offices of dentists 842558 CES6056170082 60561700 Services to buildings and dwellings 846000 CES6056170082 60561700 Services to buildings and dwellings 846000 CEU6562120057 65621200 Offices of dentists 847345 CEU6562120057 65621200 Offices of dentists 847345 CEU6562140057 65621400 Outpatient care centers 848281 CEU6562140057 65621400 Outpatient care centers 848281 CES3231100082 32311000 Food manufacturing 851406 CES3231100082 32311000 Food manufacturing 851406 CEU4142512082 41425120 Wholesale trade agents and brokers 851551 CEU4142512082 41425120 Wholesale trade agents and brokers 851551 CES5552320057 55523200 NULL 852241 CES5552320057 55523200 NULL 852241 CES4142500082 41425000 Electronic markets and agents and brokers 855291 CES4142500082 41425000 Electronic markets and agents and brokers 855291 CES5552421057 55524210 Insurance agencies and brokerages 858730 CES5552421057 55524210 Insurance agencies and brokerages 858730 CEU5051120057 50511200 Software publishers 860100 CEU5051120057 50511200 Software publishers 860100 CES6054171257 60541712 Physical 860377 CES6054171257 60541712 Physical 860377 CEU4244510082 42445100 Grocery stores 861224 CEU4244510082 42445100 Grocery stores 861224 CES5553100082 55531000 Real estate 862987 CES5553100082 55531000 Real estate 862987 CEU5553100082 55531000 Real estate 865620 CEU5553100082 55531000 Real estate 865620 CEU3133200082 31332000 Fabricated metal products 867242 CEU3133200082 31332000 Fabricated metal products 867242 CES4244510082 42445100 Grocery stores 868430 CES4244510082 42445100 Grocery stores 868430 CES3133200082 31332000 Fabricated metal products 871298 CES3133200082 31332000 Fabricated metal products 871298 CES5552420082 55524200 Insurance agencies 872398 CES5552420082 55524200 Insurance agencies 872398 CEU3133640057 31336400 Aerospace products and parts 872986 CEU3133640057 31336400 Aerospace products and parts 872986 CEU4142500082 41425000 Electronic markets and agents and brokers 873578 CEU4142500082 41425000 Electronic markets and agents and brokers 873578 CEU6054171257 60541712 Physical 874882 CEU6054171257 60541712 Physical 874882 CEU5552421057 55524210 Insurance agencies and brokerages 878065 CEU5552421057 55524210 Insurance agencies and brokerages 878065 CES3133640057 31336400 Aerospace products and parts 879785 CES3133640057 31336400 Aerospace products and parts 879785 CEU5552320057 55523200 NULL 882519 CEU5552320057 55523200 NULL 882519 CEU5552420082 55524200 Insurance agencies 894878 CEU5552420082 55524200 Insurance agencies 894878 CES5552390057 55523900 Other financial investment activities 905217 CES5552390057 55523900 Other financial investment activities 905217 CES2023700082 20237000 Heavy and civil engineering construction 908374 CES2023700082 20237000 Heavy and civil engineering construction 908374 CES4

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Create a Union table comparing average weekly earnings&nbsp; of production and nonsupervisory employees between annual_16 and january_17 using the data type 30.&nbsp; Round to the nearest penny.&nbsp; You should have a column for the average earnings and a column for the year, and the period.</span>

In [162]:
USE LaborStatisticsDB
SELECT round(avg(j.value),2) AS average_earnings, j.year, j.period
FROM dbo.january_2017 j

LEFT JOIN dbo.series s ON s.series_id = j.series_id
LEFT JOIN dbo.datatype d ON d.data_type_code = s.data_type_code

WHERE d.data_type_code = 30

GROUP BY j.year, j.period

UNION 

SELECT round(avg(a.value),2), a.year, a.period
FROM dbo.annual_2016 a

LEFT JOIN dbo.series s ON s.series_id = a.series_id
LEFT JOIN dbo.datatype d ON d.data_type_code = s.data_type_code

WHERE d.data_type_code = 30

GROUP BY a.year, a.period

(2 rows affected)

Total execution time: 00:00:00.117

average_earnings year period 797.2 2016 M13 808.53 2017 M01

## <span style="font-size:12pt;font-family:Montserrat, sans-serif;color:rgb(53, 74, 95);background-color:rgb(255, 255, 255);font-weight:700;vertical-align:baseline;white-space:pre-wrap;">Summarize Your Results</span>

<span style="font-size:10pt;font-family:Arvo, serif;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">With what you know now about the </span> <span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:&quot;Times New Roman&quot;;font-size:13.3333px;white-space:pre-wrap;">Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results and working with the</span> <span style="color:rgb(102, 102, 102);white-space:pre-wrap;font-family:&quot;Times New Roman&quot;;font-size:13.3333px;">Labor Statistics Database</span><span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:Arvo, serif;font-size:10pt;white-space:pre-wrap;">, answer the following questions. Note that while this is subjective, you should include relevant data to back up your opinion.</span>

1. <span style="background-color:rgb(255, 255, 255);font-size:10pt;white-space:pre-wrap;color:rgb(102, 102, 102);font-family:Arvo, serif;">During which time period did production and nonsupervisory employees fare better?</span>

Average weekly earnings in January 2017 were higher than average earnings Annually in 2016. 

**Avg Earnings Jan 2017** - $808.53

**Avg Earnings Annual 2016** - $797.20

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In which industries did production and nonsupervisory employees fare better?</span>

In the Annual 2016 data, the top average weekly earnings were found in the <u>Pipeline Transportation</u>**,** <u>Software Publishers</u> and **<u>Petroleum/Coal Products</u>** industries.

In the January 2017 data, the top average weekly earnings were found in the <u>Reinsurance Carriers</u>**,** <span style="color: var(--vscode-foreground);"><u style="font-weight: bold;">Petroleum/Coal Products</u> and&nbsp;</span> <span style="color: var(--vscode-foreground);"><b><u>Fossil Fuel Electric Power Generation</u></b> industries.</span>

In [163]:
USE LaborStatisticsDB
SELECT round(avg(j.value),2) AS average_earnings, j.year, j.period, i.industry_name
FROM dbo.january_2017 j

LEFT JOIN dbo.series s ON s.series_id = j.series_id
LEFT JOIN dbo.datatype d ON d.data_type_code = s.data_type_code
LEFT JOIN dbo.industry i ON i.industry_code = s.industry_code

WHERE d.data_type_code = 30

GROUP BY j.year, j.period, i.industry_name

UNION 

SELECT round(avg(a.value),2), a.year, a.period, i.industry_name
FROM dbo.annual_2016 a

LEFT JOIN dbo.series s ON s.series_id = a.series_id
LEFT JOIN dbo.datatype d ON d.data_type_code = s.data_type_code
LEFT JOIN dbo.industry i ON i.industry_code = s.industry_code

WHERE d.data_type_code = 30

GROUP BY a.year, a.period, i.industry_name

ORDER BY average_earnings desc

(1002 rows affected)

Total execution time: 00:00:00.240

average_earnings year period industry_name 1810.59 2017 M01 Reinsurance carriers 1802.43 2017 M01 Petroleum and coal products 1778.23 2017 M01 Fossil fuel electric power generation 1733.48 2017 M01 Pipeline transportation 1730.96 2016 M13 Pipeline transportation 1726.72 2017 M01 Electric power generation 1720.42 2016 M13 Software publishers 1718.26 2016 M13 Petroleum and coal products 1712.65 2016 M13 Fossil fuel electric power generation 1699.97 2016 M13 Reinsurance carriers 1689.18 2017 M01 Software publishers 1688.8 2016 M13 Electric power generation 1663.53 2017 M01 Power generation and supply 1651.11 2017 M01 Aircraft 1639.99 2016 M13 Aircraft 1637.45 2016 M13 Portfolio management 1629.16 2017 M01 Physical 1629.15 2017 M01 Custom computer programming services 1618.81 2016 M13 Custom computer programming services 1614.76 2017 M01 Computer systems design services 1608.72 2017 M01 Electric power transmission and distribution 1606.49 2017 M01 Computer systems design and related services 1596.57 2016 M13 Power generation and supply 1589.4 2017 M01 Aerospace products and parts 1587.13 2017 M01 Portfolio management 1583.98 2016 M13 Aerospace products and parts 1580.79 2016 M13 Computer systems design and related services 1570.61 2016 M13 Computer systems design services 1568.38 2016 M13 Physical 1550.03 2017 M01 Utilities 1548.21 2017 M01 Research and development in the physical 1537.03 2016 M13 Electric power transmission and distribution 1526.61 2017 M01 Oil and gas extraction 1502.24 2016 M13 Research and development in the physical 1501.5 2016 M13 Utilities 1480.1 2017 M01 Computer and software 1479.91 2017 M01 Scientific research and development services 1461.86 2016 M13 Other financial investment activities 1454.85 2017 M01 Coal mining 1445.21 2016 M13 Scientific research and development services 1443.04 2016 M13 Oil and gas pipeline construction 1439.39 2017 M01 Other financial investment activities 1437.68 2016 M13 Oil and gas extraction 1424.48 2017 M01 Other heavy construction 1419.83 2017 M01 Oil and gas pipeline construction 1411.61 2017 M01 Securities brokerage 1385.09 2016 M13 Natural gas distribution 1382.15 2017 M01 Natural gas distribution 1359.65 2017 M01 Administrative management consulting services 1354.48 2016 M13 Securities brokerage 1352.49 2017 M01 Basic chemicals 1350.5 2016 M13 Computer and software 1337.26 2016 M13 Coal mining 1331.37 2017 M01 Biotechnology research 1330.07 2017 M01 Semiconductors and related devices 1319.51 2016 M13 Administrative management consulting services 1318.75 2016 M13 Biotechnology research 1297.72 2016 M13 Investment advice 1292.87 2017 M01 Investment advice 1288.8 2017 M01 Power and communication system construction 1280.28 2016 M13 Semiconductors and related devices 1279.89 2017 M01 Architectural and engineering services 1274.39 2016 M13 Power and communication system construction 1267.89 2017 M01 Mining 1266.1 2017 M01 Offices of lawyers 1263.81 2017 M01 Wholesale trade agents and brokers 1262.33 2017 M01 Advertising agencies 1261.46 2016 M13 Architectural and engineering services 1260.63 2017 M01 Mining and logging 1260.3 2016 M13 Motor vehicle power train components 1257.53 2016 M13 Publishing industries 1255.99 2017 M01 Publishing industries 1253.58 2017 M01 Data processing 1252.66 2017 M01 Professional and technical services 1250.8 2016 M13 Other building equipment contractors 1250.64 2017 M01 Management consulting services 1250.52 2017 M01 Electronic markets and agents and brokers 1250.22 2017 M01 Management and technical consulting services 1239.51 2016 M13 Basic chemicals 1235.85 2016 M13 Utility system construction 1234.1 2017 M01 Legal services 1234.05 2016 M13 Wholesale trade agents and brokers 1233.74 2016 M13 Mining 1233.69 2016 M13 Advertising agencies 1232.42 2016 M13 Real estate credit 1232.22 2016 M13 Direct insurers 1230.99 2017 M01 Support activities for mining 1229.51 2016 M13 Offices of lawyers 1228.01 2016 M13 Steel and precast concrete contractors 1

3. <span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;"><b>Now that you have explored the datasets, is there any data or information that you wish you had in this analysis?</b></span>

 I wish the dataset contained _unique_ identifiers for Series Type/Data Type Text someplace to save from having to run the WHERE = with a text string.